src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -65,6 +65,49 @@ public static bool TryGetEnvironmentVariables(this IResource resource, [NotNullW
         return TryGetAnnotationsOfType(resource, out environmentVariables);
     }
 
+    /// <summary>
+    ///  Get the environment variables from the given resource.
+    /// </summary>
+    /// <param name="resource">The resource to get the environment variables from.</param>
+    /// <param name="applicationOperation">The context in which the AppHost is being executed</param>
+    /// <returns>The environment variables retrieved from the resource.</returns>
```

Comment:
Can we get some `<remarks>` added which cover off where this method is useful and considerations around the sync nature. As well as one or two `<example>` sections which cover off its usage.

I'm not sure this needs its own conceptual document, but the following could be expanded:

https://learn.microsoft.com/en-us/dotnet/aspire/fundamentals/testing

Could you create a docs issue to capture that?